EarlGrey/Exception/GREYDefaultFailureHandler.m

```diff
@@ -44,95 +44,76 @@ - (void)setInvocationFile:(NSString *)fileName andInvocationLine:(NSUInteger)lin
 
 - (void)handleException:(GREYFrameworkException *)exception details:(NSString *)details {
   NSParameterAssert(exception);
-  // Testcase can be nil if EarlGrey is invoked outside the context of an XCTestCase.
-  XCTestCase *currentTestCase = [XCTestCase grey_currentTestCase];
-  if (!currentTestCase) {
-    gUnknownTestExceptionCounter++;
-  }
 
-  NSMutableString *exceptionLog = [[NSMutableString alloc] init];
+  NSMutableString *log = [[NSMutableString alloc] init];
   // Start on fresh new line.
-  [exceptionLog appendString:@"\n"];
-  [exceptionLog appendFormat:@"Exception: %@\n", [exception name]];
-  if ([exception reason]) {
-    [exceptionLog appendFormat:@"Reason: %@\n", [exception reason]];
-  } else {
-    [exceptionLog appendString:@"Reason for exception was not provided.\n"];
-  }
+  [log appendString:@"\n"];
+  [log appendFormat:@"Exception: %@\n", exception.name];
+  NSString *reason = exception.reason ? exception.reason : @"exception.reason was not provided";
+  [log appendFormat:@"Reason: %@\n", reason];
   if (details.length > 0) {
-    [exceptionLog appendFormat:@"%@\n", details];
+    [log appendFormat:@"%@\n", details];
   }
-  [exceptionLog appendString:@"\n"];
+  [log appendString:@"\n"];
 
+  // Test case can be nil if EarlGrey is invoked outside the context of an XCTestCase.
+  XCTestCase *currentTestCase = [XCTestCase grey_currentTestCase];
   NSString *screenshotName;
   if (currentTestCase) {
     screenshotName = [NSString stringWithFormat:@"%@_%@",
                                                 [currentTestCase grey_testClassName],
                                                 [currentTestCase grey_testMethodName]];
   } else {
-    screenshotName =
-        [NSString stringWithFormat:@"unknown_%lu", (unsigned long)gUnknownTestExceptionCounter];
+    gUnknownTestExceptionCounter++;
```

Comment:
Although this may shorten the code we have added this here to keep code that modifies the global separate for everything else.